contrib/epee/src/hex.cpp

```diff
@@ -84,7 +86,42 @@ namespace epee
     return write_hex(out, src);
   }
 
-  std::vector<uint8_t> from_hex::vector(boost::string_ref src)
+
+  bool from_hex::to_string(std::string& out, const boost::string_ref src)
+  {
+    out.resize(src.size() / 2);
+    return to_buffer_unchecked(reinterpret_cast<std::uint8_t*>(&out[0]), src);
+  }
+
+  bool from_hex::to_buffer(span<std::uint8_t> out, const boost::string_ref src) noexcept
+  {
+    if (src.size() / 2 != out.size())
+      return false;
+    return to_buffer_unchecked(out.data(), src);
+  }
+
+  bool from_hex::to_buffer_unchecked(std::uint8_t* dst, const boost::string_ref s) noexcept
+  {
+      if (s.size() % 2 != 0)
+        return false;
+
+      const unsigned char *src = (const unsigned char *)s.data();
+      for(size_t i = 0; i < s.size(); i += 2)
+      {
+        int tmp = *src++;
+        tmp = epee::misc_utils::parse::isx[tmp];
+        if (tmp == 0xff) return false;
+        int t2 = *src++;
+        t2 = epee::misc_utils::parse::isx[t2];
+        if (t2 == 0xff) return false;
+        *dst++ = (tmp << 4) | t2;
+      }
+
+      return true;
+  }
+
+
+  std::vector<uint8_t> from_hex_locale::to_vector(const boost::string_ref src)
```

Comment:
I can't comment on the offending line, because it already exists in master apparently, but this code assumes that a `boost::string_ref` is always null-terminated, which it doesn't necessarily have to be.

The function contains the following incorrect comment:

> // the data to work with (std::string is always null-terminated)

I am unsure whether we should fix it in this PR or whether a separate fix-PR should be made instead. I tend to lean towards fixing it here, since fixing it in another PR would likely yield merge conflicts when this PR gets merged first.

Reply:
I noticed this as well but wanted to do it in a separate PR so that reviewers would focus just on that.

Reply:
See PR #6359.